src/main/resources/assets/computercraft/lua/rom/programs/list.lua

```diff
@@ -31,8 +50,33 @@ end
 table.sort( tDirs )
 table.sort( tFiles )
 
-if term.isColour() then
-    textutils.pagedTabulate( colors.green, tDirs, colors.white, tFiles )
+local function writeFileTable( tPrint, tList )
+    for i, v in ipairs( tList ) do
+        tAttr = fs.attributes( shell.resolve( v ) )
+        tSingeEntry = {v}
+        if tAttr["isDir"] then
+            table.insert( tSingeEntry, "Dir" )
+        else
+            table.insert( tSingeEntry, "File" )
+        end
+        sDate = settings.get("list.date_format")
+        table.insert( tSingeEntry, tostring( tAttr["size"] ) )
+        table.insert( tSingeEntry, os.date( sDate, tAttr["created"] ) )
+        table.insert( tSingeEntry, os.date( sDate, tAttr["modification"] ) )
+        table.insert( tPrint, tSingeEntry )
+    end
+    return tPrint
+end
+
+if bShowAttr then
+    tPrint = {{"Name","Type","Size","Created","Modified"}}
+    tPrint = writeFileTable( tPrint, tDirs )
```

Comment:
I wouldn't bother to return `tPrint` in `writeFileTable`. `writeFileTable( tPrint, tDirs )` ends up being cleaner than `tPrint = writeFileTable( tPrint, tDirs )`.